### PR TITLE
Add GRU classifier and GRU time series model

### PR DESCRIPTION
**Pull request by Derek-Wds**
Hi, I have added two GRU models in this PR :smile: . Since, the architectures of the two LSTM models used in this repo are reasonable, I just wrote the GRU models based on them with simple modifications. I have also provided the unit test codes, and all tests passed on my machine.

If there are any problems or additional things to do, please let me know. Hope it helps!

Derek

### CODE REVIEW

**Review by sneaxiy — COMMENTED** [review]
Why not using `WITH` statements to choose `LSTM` or `GRU` model, instead of copying most of the model codes from LSTM-based models?

**Comment by Derek-Wds** [comment]
Yes, that is definitely a better solution. If this solution makes sense, we could even combine the RNN family models (RNN, LSTM, GRU) into one class. At the same time, we could give the user freedom to choose whether using the `Bidirectional` option in the model. 

If this sounds good, I can update my PR to make an initial version of it.

**Comment by typhoonzero** [comment]
> Yes, that is definitely a better solution. If this solution makes sense, we could even combine the RNN family models (RNN, LSTM, GRU) into one class. At the same time, we could give the user freedom to choose whether using the Bidirectional option in the model.

Sounds cool.

**Comment by Derek-Wds** [comment]
Hi, I have just updated the codes for RNN classifier and RNN TS model, where I combine vanilla RNN, LSTM and GRU into one model respectively. Hope this is readable and more efficient.

File: sqlflow_models/rnn_based_time_series.py [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
+        :param stack_units: Units for RNN layer.
             type stack_units: vector of ints.
         :param n_in: Size of time window.
             type n_in: int.
         :param n_out: Number of predicted labels.
             type n_out: int.
         :param n_features: number of features in every time window.
-            type n_features: int
+            type n_features: int.
+        :param model_type: Specific RNN model to be used.
```
Comment (typhoonzero) [inline sqlflow_models/rnn_based_time_series.py]: Add a description about possible values, e.g. "rnn", "lstm" and "gru"?

File: sqlflow_models/rnnclassifier.py [hunk 2]
```diff
@@ -2,32 +2,38 @@
...
+    def __init__(self, feature_columns=None, stack_units=[32], hidden_size=64, n_classes=2, model_type='rnn', bidirectional=False):
+        """StackedRNNClassifier
         :param feature_columns: All columns must be embedding of sequence column with same sequence_length.
         :type feature_columns: list[tf.embedding_column].
-        :param stack_units: Units for LSTM layer.
+        :param stack_units: Units for RNN layer.
         :type stack_units: vector of ints.
         :param n_classes: Target number of classes.
         :type n_classes: int.
+        :param model_type: Specific RNN model to be used.
```
Comment (typhoonzero) [inline sqlflow_models/rnnclassifier.py]: Same as above.

**Review by typhoonzero — COMMENTED** [review]
LGTM generally. Please be aware that the model name `LSTMBasedTimeSeriesModel` is already used in SQLFlow's unit test: https://github.com/sql-machine-learning/sqlflow/blob/c58f0a4f6b08984c667d3ace1aa21d09aa6112de/cmd/sqlflowserver/e2e_mysql_test.go#L194 and the tutorial https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/tutorial/energe_lstmbasedtimeseries.md. Can you pleas also update those two places after this PR was merged?

**Comment by Derek-Wds** [comment]
Hi, I have just added one sentence for the description of the model type. Please take a look.

> LGTM generally. Please be aware that the model name `LSTMBasedTimeSeriesModel` is already used in SQLFlow's unit test: https://github.com/sql-machine-learning/sqlflow/blob/c58f0a4f6b08984c667d3ace1aa21d09aa6112de/cmd/sqlflowserver/e2e_mysql_test.go#L194 and the tutorial https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/tutorial/energe_lstmbasedtimeseries.md. Can you pleas also update those two places after this PR was merged?

Yeah, sure! Once this PR is being merged, I will try my best to modify the unit tests and tutorials in two days.

**Review by typhoonzero — APPROVED** [review]
LGTM

**Pull request merged**